pytest_html/plugin.py

```diff
@@ -83,97 +84,112 @@ def __init__(self, logfile):
-            additional_html.append(html.div(
-                html.a(html.img(src=image), href="#"),
-                class_='image'))
-
-        elif extra.get('format') == extras.FORMAT_HTML:
-            additional_html.append(html.div(raw(extra.get('content'))))
-
-        elif extra.get('format') == extras.FORMAT_JSON:
-            href = data_uri(json.dumps(extra.get('content')),
-                            mime_type='application/json')
-
-        elif extra.get('format') == extras.FORMAT_TEXT:
-            href = data_uri(extra.get('content'))
-
-        elif extra.get('format') == extras.FORMAT_URL:
-            href = extra.get('content')
-
-        if href is not None:
-            links_html.append(html.a(
-                extra.get('name'),
-                class_=extra.get('format'),
-                href=href,
-                target='_blank'))
-            links_html.append(' ')
-
-    def append_log_html(self, report, additional_html):
-        log = html.div(class_='log')
-        if report.longrepr:
-            for line in str(report.longrepr).splitlines():
-                if not PY3:
-                    line = line.decode('utf-8')
-                separator = line.startswith('_ ' * 10)
-                if separator:
-                    log.append(line[:80])
-                else:
-                    exception = line.startswith("E   ")
-                    if exception:
-                        log.append(html.span(raw(escape(line)),
-                                             class_='error'))
+    class TestResult:
+
+        def __init__(self, outcome, report):
+            self.test_id = report.nodeid
+            if report.when != 'call':
+                self.test_id = '::'.join([report.nodeid, report.when])
+            self.time = getattr(report, 'duration', 0.0)
+            self.outcome = outcome
+            self.additional_html = []
+            self.links_html = []
+
+            self.append_log_html(report, self.additional_html)
+
+            for extra in getattr(report, 'extra', []):
+                self.append_extra_html(extra,
+                                       self.additional_html, self.links_html)
+
+            self.row_table = html.tr([
+                html.td(self.outcome, class_='col-result'),
+                html.td(self.test_id, class_='col-name'),
+                html.td('{0:.2f}'.format(self.time), class_='col-duration'),
+                html.td(self.links_html, class_='col-links')])
+
+            self.row_extra = html.tr(html.td(self.additional_html,
+                                     class_='extra', colspan='5'))
+
+        def __lt__(self, other):
+            order = ('Error', 'Failed', 'Rerun', 'XFailed',
+                     'XPassed', 'Skipped', 'Passed')
+            if order.index(self.outcome) >= order.index(other.outcome):
+                return False
+            elif order.index(self.outcome) < order.index(other.outcome):
+                return True
```

Comment:
This can be simplified further as:

```python
def __lt__(self, other):
    order = ('Error', 'Failed', 'Rerun', 'XFailed', 'XPassed', 'Skipped', 'Passed')
    return order.index(self.outcome) < order.index(other.outcome)
```